assembly/src/parsers/context.rs

```diff
@@ -255,7 +286,11 @@ impl ParserContext {
         };
 
         // parse procedure body
-        let body = self.parse_body(tokens, false)?;
+        let ParsedCode {
+            nodes,
+            mut locations,
+        } = self.parse_body(tokens, false)?;
+        locations.insert(0, header_location);
```

Comment:
I think this will make mapping nodes to locations a bit more complicated (they will be all offset by one). So, I'm wondering if it would be better to have `header_location` as a separate field in the the `ProcedureAst` struct.